handler/engine/src/main/java/io/knotx/fragments/engine/EventLogEntry.java

```diff
@@ -26,35 +27,42 @@
   private static final String STATUS_KEY = "status";
   private static final String TRANSITION_KEY = "transition";
   private static final String TIMESTAMP_KEY = "timestamp";
+  private static final String ACTION_LOG_KEY = "actionLog";
 
-  private String task;
-  private String action;
-  private ActionStatus status;
-  private String transition;
-  private long timestamp;
+  private final String task;
+  private final String action;
+  private final ActionStatus status;
+  private final String transition;
+  private final long timestamp;
+  private final JsonObject actionLog;
 
-  public static EventLogEntry success(String task, String action, String transition) {
-    return new EventLogEntry(task, action, ActionStatus.SUCCESS, transition);
+  public static EventLogEntry success(String task, String action, FragmentResult fragmentResult) {
+    return new EventLogEntry(task, action, ActionStatus.SUCCESS, fragmentResult.getTransition(), fragmentResult.getActionLog());
   }
 
   public static EventLogEntry unsupported(String task, String action, String transition) {
-    return new EventLogEntry(task, action, ActionStatus.UNSUPPORTED_TRANSITION, transition);
+    return new EventLogEntry(task, action, ActionStatus.UNSUPPORTED_TRANSITION, transition,null);
+  }
+
+  public static EventLogEntry error(String task, String action, String transition, JsonObject actionLog) {
+    return new EventLogEntry(task, action, ActionStatus.ERROR, transition, actionLog);
   }
 
   public static EventLogEntry error(String task, String action, String transition) {
```

Comment:
Why make possible to create error without `actionLog`?

Reply:
It is for Composite Node. If one component of Composite fails it is handled by 
io.knotx.fragments.engine.TaskExecutionContext#handleError. Action log data are passed.

Then io.knotx.fragments.engine.TaskExecutionContext#toFragmentResult is called. We have no info about action Log here for components of Composite. As mentions it is handled earlier.